src/scan_to_cloud_filter_chain.cpp

```diff
@@ -95,8 +107,9 @@ class ScanToCloudFilterChain
   bool  incident_angle_correction_;
 
   ////////////////////////////////////////////////////////////////////////////////
-  ScanToCloudFilterChain () : laser_max_range_ (DBL_MAX), private_nh("~"), filter_(tf_, "", 50),
-                   cloud_filter_chain_("sensor_msgs::PointCloud2"), scan_filter_chain_("sensor_msgs::LaserScan")
+  ScanToCloudFilterChain (ros::NodeHandle& nh, const std::string& name) :
+      laser_max_range_ (DBL_MAX), private_nh(nh), name_(name), filter_(tf_, "", 50),
```

Comment:
Could you change the parameter name that gets passed in to the contructor from `nh` to `pnh`? Usually `nh` is the public node handle and `pnh` is the private node handle, they exist side by side, but have different values.